lib/RGBot.js

```diff
@@ -932,101 +1654,167 @@ const RGBot = class {
         return result;
     }
 
+    /**
+     * Opens a container, then perform the supplied useFunction before closing the container.
+     * This function does NOT approach the container.  It must already be in reach of the bot
+     *
+     * @param {Block} containerBlock
+     * @param {object} [options={}]
+     * @param {function(openContainer:Window,options:object):Promise<boolean>} useContainerFunction
+     * @returns {Promise<boolean>}
+     */
+    async openAndUseContainer( containerBlock, options={}, useContainerFunction= async (openContainer, options) => {}) {
```

Comment:
This is one of the only functions we have that takes a function to perform some operation. Since the last operation that happens is that function, why not just have the user do `await openContainer(); await useContainerFunction()`? Is it to properly close the container if the user's function throws an error?

Reply:
I also /hate this pattern btw...  The reason I made it this way was twofold
1. Primarily to spark some discussion about this concept in general
2. Because while I found uses for deposit and withdraw, someone else may want to provide some customer deposit logic that manages stacks, or some annoying logic that files every slot in the container with a single dirt block to screw up other players' bots.  Unfortunately, we really need them to follow the pattern of managing the `isLookingInContainer` so we don't wrongfully mark their bot as stuck.  This pattern somewhat forces that on them, but I agree.. is non-obvious and would definitely not lend itself to block programming or other similar techniques.

I think the solution is that `openAndUseContainer` becomes hidden/private and the `deposit`/`withdraw` functions take in a containerBlock (not containerWindow) and utilize `openAndUseContainer` internally.

I definitely do not like the pattern of `await openChest` `await depositItems` for an implementer.  They don't need to know about mineflayer needing to do open and action as 2 steps based as limited by the model types.

Reply:
I like your suggestion, if I understand it correctly.